hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/queryplan/physical/impl/PhysicalOpBindJoinWithBoundJoin.java

```diff
@@ -76,4 +84,49 @@ public boolean equals( final Object o ) {
 	public String toString() {
 		return "> BoundJoinBindJoin" + lop.toString();
 	}
+
+	public static class Factory implements PhysicalOpFactory
+	{
+		@Override
+		public boolean supports( final LogicalOperator lop, final ExpectedVariables inputVars ) {
+			if ( lop instanceof LogicalOpGPAdd op ) {
+				return isSupported( op.getFederationMember(), op.getPattern(), inputVars );
+			}
+			if ( lop instanceof LogicalOpGPOptAdd op ) {
+				return isSupported( op.getFederationMember(), op.getPattern(), inputVars );
+			}
+			return false;
+		}
+
+		@Override
+		public PhysicalOperator create( final LogicalOperator lop ) {
+			if ( lop instanceof LogicalOpGPAdd op ) {
+				return new PhysicalOpBindJoinWithBoundJoin(op);
+			}
+			else if ( lop instanceof LogicalOpGPOptAdd op ) {
+				return new PhysicalOpBindJoinWithBoundJoin(op);
+			}
+
+			throw new UnsupportedOperationException( "Unsupported type of logical operator: " + lop.getClass().getName() + "." );
+		}
+
+		private boolean isSupported( final FederationMember fm,
+		                             final SPARQLGraphPattern pattern,
+		                             final ExpectedVariables vars )
+		{
+			return (fm instanceof SPARQLEndpoint) && hasNonJoiningVar(pattern, vars);
+		}
+
+		private boolean hasNonJoiningVar( final SPARQLGraphPattern pattern, final ExpectedVariables vars ) {
+			final Set<Var> certainVars = vars != null ? vars.getCertainVariables() : Set.of();
+			final Set<Var> possibleVars = vars != null ? vars.getPossibleVariables() : Set.of();
```

Comment:
To avoid creating Java objects for the empty set in the `vars == null` case, it would be better to handle this case first:
```suggestion
			if ( vars == null )
				return true;

			final Set<Var> certainVars = vars.getCertainVariables();
			final Set<Var> possibleVars = vars.getPossibleVariables();
```